// my-core/MyLogger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var MyLogger = /** @class */ (function () {
    function MyLogger() {
        this._log = true;
    }
    MyLogger.prototype.silence = function (flag) {
        this._log = !flag;
    };
    MyLogger.prototype.log = function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        if (!this._log) {
            return;
        }
        console.log.call(console, "[INF] [" + new Date() + "]:", args.join(' '));
    };
    MyLogger.prototype.logForce = function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        console.log.call(console, "[INF] [" + new Date() + "]:", args.join(' '));
    };
    MyLogger.prototype.error = function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        console.log.call(console, "[ERR] [" + new Date() + "]:", args.join(' '));
    };
    return MyLogger;
}());
exports.MyLogger = MyLogger;
var EmptyLogger = /** @class */ (function () {
    function EmptyLogger() {
    }
    EmptyLogger.prototype.silence = function (flag) {
    };
    EmptyLogger.prototype.log = function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
    };
    EmptyLogger.prototype.logForce = function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
    };
    EmptyLogger.prototype.error = function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
    };
    return EmptyLogger;
}());
exports.EmptyLogger = EmptyLogger;
